### PR TITLE
Update 1-2 to follow BASHER

### PR DESCRIPTION
**Pull request by derisen**
## Purpose

* Update project to follow BASHER
* Update all dependencies
* Regenerate the README

## Does this introduce a breaking change?

```
[ x ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ x ] Code style update (formatting, local variables)
[ x ] Refactoring (no functional changes, no api changes)
[ x ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 1-Authentication/2-sign-in-b2c/AppCreationScripts/sample.json [hunk 1]
```diff
@@ -1,42 +1,49 @@
...
+    },
+    "AADApps": [
         {
-          "Comment": "Navigate to the Manifest page, find the 'replyUrlsWithType' section and change the type of redirect URI to 'Spa'"
+            "Id": "client",
+            "Name": "msal-angular-spa",
+            "Kind": "SinglePageApplication",
+            "Audience": "AzureADandPersonalMicrosoftAccount",
+            "HomePage": "http://localhost:4200",
+            "ReplyUrls": "http://localhost:4200, http://localhost:4200/auth"
```
Comment (salman90) [inline 1-Authentication/2-sign-in-b2c/AppCreationScripts/sample.json:29]: Add SampleSubPath

File: 1-Authentication/2-sign-in-b2c/SPA/src/app/auth-config.ts [hunk 2]
```diff
@@ -59,11 +66,22 @@ export const b2cPolicies = {
...
+ */
+export const loginRequest = {
+    scopes: []
+}
+
 /**
  * An optional silentRequest object can be used to achieve silent SSO
- * between applications by providing a "login_hint" property.
+ * between applications by providing a "loginHint" property. For more, visit:
+ * https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-js-sso#sso-between-different-apps
```
Comment (salman90) [inline 1-Authentication/2-sign-in-b2c/SPA/src/app/auth-config.ts:82]: Add loginHint steps in the readme

**Comment by v-michaelmi** [comment]
There's still an old `isIE` reference in `auth-config.ts`. We should be able to retire it now?

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 3]
```diff
@@ -1,24 +1,44 @@
...
- 1. [Contributing](#contributing)
+* [Overview](#overview)
+* [Scenario](#scenario)
+* [Contents](#contents)
+* [Prerequisites](#prerequisites)
+* [Explore the sample](#explore-the-sample)
+* [Troubleshooting](#troubleshooting)
+* [About the code](#about-the-code)
+* [Next Steps](#next-steps)
+* [Contributing](#contributing)
```
Comment (salman90) [inline 1-Authentication/2-sign-in-b2c/README.md:32]: Learn more and step up the sample titles are missing.

File: 1-Authentication/2-sign-in-b2c/SPA/package.json [hunk 4]
```diff
@@ -1,50 +1,42 @@
...
-  "description": "Angular single-page application using MSAL React to sign-in users against Azure AD B2C",
+  "name": "spa",
+  "version": "0.0.0",
   "scripts": {
     "ng": "ng",
     "start": "ng serve",
     "build": "ng build",
-    "test": "ng test --watch=false --no-progress --browsers=ChromeHeadlessCI",
-    "lint": "ng lint",
-    "e2e": "ng e2e"
```
Comment (v-michaelmi) [inline 1-Authentication/2-sign-in-b2c/SPA/package.json:11]: Do we want this command back? There's some documentation for using this.

File: 1-Authentication/2-sign-in-b2c/SPA/src/app/app.component.ts [hunk 5]
```diff
@@ -2,86 +2,202 @@ import { Component, OnInit, Inject, OnDestroy } from '@angular/core';
...
+
+        this.msalBroadcastService.msalSubject$
+            .pipe(
+                filter((msg: EventMessage) => msg.eventType === EventType.LOGIN_FAILURE || msg.eventType === EventType.ACQUIRE_TOKEN_FAILURE),
+                takeUntil(this._destroying$)
+            )
+            .subscribe((result: EventMessage) => {
+                // Checking for the forgot password error. Learn more about B2C error codes at
+                // https://learn.microsoft.com/azure/active-directory-b2c/error-codes
+                if (result.error && result.error.message.indexOf('AADB2C90118') > -1) {
```
Comment (v-michaelmi) [inline 1-Authentication/2-sign-in-b2c/SPA/src/app/app.component.ts:133]: Nit:

`!!result.error.message.find('AADB2C90118')`

File: 1-Authentication/2-sign-in-b2c/SPA/src/app/app.component.ts [hunk 6]
```diff
@@ -2,86 +2,202 @@ import { Component, OnInit, Inject, OnDestroy } from '@angular/core';
...
+                        scopes: [],
+                    };
+
+                    this.login(resetPasswordFlowRequest);
+                };
+            });
+    }
+
+    setLoginDisplay() {
+        this.loginDisplay = this.authService.instance.getAllAccounts().length > 0;
```
Comment (v-michaelmi) [inline 1-Authentication/2-sign-in-b2c/SPA/src/app/app.component.ts:145]: Nit, magic number:

`this.loginDisplay = !!this.authService.instance.getAllAccounts().length;`

File: 1-Authentication/2-sign-in-b2c/SPA/src/app/app.component.ts [hunk 7]
```diff
@@ -2,86 +2,202 @@ import { Component, OnInit, Inject, OnDestroy } from '@angular/core';
...
+
+    checkAndSetActiveAccount() {
+        /**
+         * If no active account set but there are accounts signed in, sets first account to active account
+         * To use active account set here, subscribe to inProgress$ first in your component
+         * Note: Basic usage demonstrated. Your app may require more complicated account selection logic
+         */
+        let activeAccount = this.authService.instance.getActiveAccount();
+
+        if (!activeAccount && this.authService.instance.getAllAccounts().length > 0) {
```
Comment (v-michaelmi) [inline 1-Authentication/2-sign-in-b2c/SPA/src/app/app.component.ts:156]: `!!this.authService.instance.getAllAccounts().length`

File: 1-Authentication/2-sign-in-b2c/SPA/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# SPA
```
Comment (salman90) [inline 1-Authentication/2-sign-in-b2c/SPA/README.md:1]: Do we need this readme?

**Comment by v-michaelmi** [comment]
A couple of nits otherwise LGTM

**Review by salman90 — APPROVED** [review]
LGTM

File: 1-Authentication/2-sign-in-b2c/SPA/src/app/auth-config.ts (hunk 2 above)
Comment (derisen) [reply 1-Authentication/2-sign-in-b2c/SPA/src/app/auth-config.ts:82]: Good point -will need to check if this applies to B2C :/

File: 1-Authentication/2-sign-in-b2c/SPA/package.json (hunk 4 above)
Comment (derisen) [reply 1-Authentication/2-sign-in-b2c/SPA/package.json:11]: Come again? 😃

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: 1-Authentication/2-sign-in-b2c/AppCreationScripts/sample.json (hunk 1 above)
Comment (derisen) [reply 1-Authentication/2-sign-in-b2c/AppCreationScripts/sample.json:29]: done

File: 1-Authentication/2-sign-in-b2c/README.md (hunk 3 above)
Comment (derisen) [reply 1-Authentication/2-sign-in-b2c/README.md:33]: done

File: 1-Authentication/2-sign-in-b2c/SPA/README.md (hunk 8 above)
Comment (derisen) [reply 1-Authentication/2-sign-in-b2c/SPA/README.md:1]: removed

File: 1-Authentication/2-sign-in-b2c/SPA/src/app/auth-config.ts (hunk 2 above)
Comment (derisen) [reply 1-Authentication/2-sign-in-b2c/SPA/src/app/auth-config.ts:82]: added link to show how to enable to username in id token (which can be used as loginHint later)

**Pull request merged**